src/main/java/uk/gov/companieshouse/api/accounts/validation/CreditorsWithinOneYearValidator.java

```diff
@@ -0,0 +1,191 @@
+                creditorsPreviousPeriod.getOtherCreditors() != null) &&
+                (creditorsPreviousPeriod.getTotal() == null)) {
+
+            addError(errors, invalidNote, CREDITORS_WITHIN_PREVIOUS_PERIOD_TOTAL_PATH);
+        }
+    }
+
+    private void validatePreviousTotalCalculationCorrect(@Valid PreviousPeriod creditorsPreviousPeriod, Errors errors) {
+        if (creditorsPreviousPeriod.getTotal() != null) {
+
+            Long bankLoans =
+                    Optional.ofNullable(creditorsPreviousPeriod.getBankLoansAndOverdrafts()).orElse(0L);
+            Long amountsDueUnderFinance =
+                    Optional.ofNullable(creditorsPreviousPeriod.getFinanceLeasesAndHirePurchaseContracts()).orElse(0L);
+            Long tradeCreditors =
+                    Optional.ofNullable(creditorsPreviousPeriod.getTradeCreditors()).orElse(0L);
+            Long taxation =
+                    Optional.ofNullable(creditorsPreviousPeriod.getTaxationAndSocialSecurity()).orElse(0L);
+            Long accruals =
+                    Optional.ofNullable(creditorsPreviousPeriod.getAccrualsAndDeferredIncome()).orElse(0L);
+            Long otherCreditors =
+                    Optional.ofNullable(creditorsPreviousPeriod.getOtherCreditors()).orElse(0L);
+
+            Long total = creditorsPreviousPeriod.getTotal();
+            Long sum =
+                    bankLoans + amountsDueUnderFinance + tradeCreditors + taxation + accruals + otherCreditors;
+
+            validateAggregateTotal(total, sum,
+                    CREDITORS_WITHIN_PREVIOUS_PERIOD_TOTAL_PATH, errors);
+        }
+    }
+
+    private void validateCurrentPeriod(@Valid CurrentPeriod creditorsCurrentPeriod,
+            Errors errors) {
+
+        validateTotalFieldPresentIfCurrentValuesProvided(creditorsCurrentPeriod, errors);
+        validateCurrentPeriodTotalCalculation(creditorsCurrentPeriod, errors);
+        validateDetailsPresentIfNoTotalProvided(creditorsCurrentPeriod, errors);
+    }
+
+    private void validateTotalFieldPresentIfCurrentValuesProvided(@Valid CurrentPeriod creditorsCurrentPeriod, Errors errors) {
+        if ((creditorsCurrentPeriod.getBankLoansAndOverdrafts() != null ||
+                creditorsCurrentPeriod.getFinanceLeasesAndHirePurchaseContracts() != null ||
+                creditorsCurrentPeriod.getTradeCreditors() != null ||
+                creditorsCurrentPeriod.getTaxationAndSocialSecurity() != null ||
+                creditorsCurrentPeriod.getAccrualsAndDeferredIncome() != null ||
+                creditorsCurrentPeriod.getOtherCreditors() != null) &&
+                creditorsCurrentPeriod.getTotal() == null) {
+
+            addError(errors, invalidNote, CREDITORS_WITHIN_CURRENT_PERIOD_TOTAL_PATH);
+        }
+    }
+
+    private void validateDetailsPresentIfNoTotalProvided(@Valid CurrentPeriod creditorsCurrentPeriod, Errors errors) {
+        if (creditorsCurrentPeriod.getBankLoansAndOverdrafts() == null &&
```

Comment:
From the method name, I'd expect the logic to look like...
```
if (creditorsCurrentPeriod.getTotal() == null && creditorsCurrentPeriod.getDetails() == null) {
   // add error here
}
```
Previous methods should be sufficient to validate numeric fields.

Question is, should we ever accept a `null` total?